### PR TITLE
feat: add Go and Python to linting guideline

### PR DESCRIPTION
**Pull request by grayside**
Furthers #102

This is a combined update for Python & Go, so it needs LGTM from members of each language team.

### CODE REVIEW

**Review by iennae — CHANGES_REQUESTED** [review]
I think this needs additional context. I literally went to the URLs mentioned and I couldn't find specifically linting guidance from the front page for Go. If we send folks to a different page we should really make sure it has the guidance they need.

**Comment by iennae** [comment]
Python does have linting guidance, but we should still say a little more context. 

Something like 

Follow the guidance as per the Python style guide on linting https://google.github.io/styleguide/pyguide#21-lint with the recommended lint configuration https://google.github.io/styleguide/pylintrc (is that actually our recommendation) 

For Go, I don't know what we're trying to point to here.

**Review by telpirion — APPROVED** [review]

**Comment by telpirion** [comment]
> I think this needs additional context. I literally went to the URLs mentioned and I couldn't find specifically linting guidance from the front page for Go. If we send folks to a different page we should really make sure it has the guidance they need.

That really is the best starting point for "official" Go guidance, with the other potential source being [Effective Go](https://go.dev/doc/effective_go).

**Comment by grayside** [comment]
More specifically, you can preview this change at https://googlecloudplatform.github.io/samples-style-guide/previews/PR-127/#linting

File: content/en/_index.md [hunk 1]
```diff
@@ -946,7 +946,20 @@ Follow the [Google Java Style Guide](https://google.github.io/styleguide/javagui
...
 {{% /tab %}}
 
+{{% tab header="Go" text=true %}}
+Follow the [Google Go Style Guide](https://google.github.io/styleguide/go/) and [Effective Go](https://go.dev/doc/effective_go).
+
+The official linters for go are [gofmt](https://pkg.go.dev/cmd/gofmt), [go vet](https://pkg.go.dev/cmd/vet), and [go lint](https://github.com/golang/lint).
+{{% /tab %}}
+
+{{% tab header="Python" text=true %}}
+All Python samples should follow the best practices defined in the [PEP 8](https://www.python.org/dev/peps/pep-0008/) style guide and the [Google Python Style Guide](http://google.github.io/styleguide/pyguide.html).
```
Comment (engelke) [inline content/en/_index.md:956]: Remove the Google Python Style Guide. It conflicts with our standard style (such as indentation amount). For public samples, we follow the public Python PEP guide.

**Review by engelke — CHANGES_REQUESTED** [review]
Please remove Google style guide reference. See comment.

File: content/en/_index.md (hunk 1 above)
Comment (grayside) [inline content/en/_index.md:956]:
```suggestion
All Python samples should follow the best practices defined in the [PEP 8](https://www.python.org/dev/peps/pep-0008/) style guide.
```

**Review by engelke — APPROVED** [review]
LGTM

**Review by iennae — APPROVED** [review]
LGTM

**Pull request merged**